moon/controller.py

```diff
@@ -328,14 +320,17 @@ def try_step_around(self):
         self.go_straight(3.0, timeout=timedelta(seconds=20))
         self.turn(math.radians(-90), timeout=timedelta(seconds=10))
 
-    def run(self):
+    def wait_for_init(self):
         try:
             print('Wait for definition of last_position and yaw')
-            while self.last_position is None or self.yaw is None:
-                self.update()  # define self.sim_time
+            while self.sim_time is None or self.last_position is None or self.yaw is None:
+                self.update()  # define self.time
```

Comment:
the older comment was right or not? (it updates both sim_time (in which we are interested) as well as time